anesthetic/utils.py

```diff
@@ -387,6 +387,8 @@ def sample_compression_1d(x, w=None, n=1000):
     x, w, array-like
         Compressed samples and weights
     """
+    if n is None:
+        return x, w
```

Comment:
This change to the default differs from `compress_weights` and therefore `WeightedDataFrame.compress`. There, if `n=None` it compresses to the channel capacity. I see the logic of `None` being 'do not compress', but equally there's something convenient about calling `.compress()` and it compressing optimally. Whatever we choose, we should be consistent across compression functions.

Reply:
Should we expand with a `bool` option? `True` for channel capacity, `False` for no compression?

Reply:
I like that -- could we have that convention for all compression functions?

Reply:
implemented in [aba18fa](https://github.com/williamjameshandley/anesthetic/pull/216/commits/aba18fa7b048c5aad7251aa5b2d0caf4248c4303)